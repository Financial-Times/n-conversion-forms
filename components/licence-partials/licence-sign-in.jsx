import React from 'react';
import PropTypes from 'prop-types';

export function LicenceSignIn ({
	displayName = null,
	url = null,
}) {
	return (
		<div className="ncf__center">
			<p>
				{'Have you already joined the '}
				{displayName && (<span className="bold-licence-text">displayName</span>)}
				{' account?'}
			</p>
			{url &&
                (<div>
                	<a href={url} className="ncf__button--secondary mid-width-button">Sign in</a>
                </div>)
			}
		</div>
	);
}

LicenceSignIn.propTypes = {
	displayName: PropTypes.string,
	url: PropTypes.string
};