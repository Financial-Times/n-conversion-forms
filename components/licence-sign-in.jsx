import React from 'react';
import PropTypes from 'prop-types';

export function LicenceSignIn ({
	displayName = '',
	url = '',
}) {
	return (
		<div className="ncf__center">
			{
				renderSignInTitle(displayName)
			}
			{(url !== '') &&
				(<div>
					<a href={url} className="ncf__button--secondary ncf__button--mid-width">Sign in</a>
				</div>)
			}
		</div>
	);
}

function renderSignInTitle (displayName) {
	let signInTitle;

	if (displayName !== '') {
		signInTitle =
			<p>
				<span>Have you already joined the </span>
				<span className="ncf__bold-licence-text">{displayName}</span>
				<span> account?</span>
			</p>;
	} else {
		signInTitle = <p>Already have a subscription?</p>;
	}

	return signInTitle;
}

LicenceSignIn.propTypes = {
	displayName: PropTypes.string,
	url: PropTypes.string
};
