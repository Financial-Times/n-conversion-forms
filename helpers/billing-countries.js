import { billingCountries } from 'n-common-static-data';

export { billingCountries };
