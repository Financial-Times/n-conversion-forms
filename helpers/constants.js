const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
