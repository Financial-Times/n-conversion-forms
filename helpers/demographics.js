const { demographics } = require('n-common-static-data');

export { demographics };
