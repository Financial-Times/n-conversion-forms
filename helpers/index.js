module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
