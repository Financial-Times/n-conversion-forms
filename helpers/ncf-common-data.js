const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
