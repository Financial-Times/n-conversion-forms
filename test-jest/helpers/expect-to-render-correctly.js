import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
