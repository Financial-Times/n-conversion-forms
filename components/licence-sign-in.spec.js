import { expectToRenderCorrectly } from '../test-jest/helpers/expect-to-render-correctly';
import { LicenceSignIn } from './licence-sign-in';

expect.extend(expectToRenderCorrectly);

describe('LicenceSignIn', () => {
	it('renders with default props', () => {
		const props = {};

		expect(LicenceSignIn).toRenderCorrectly(props);
	});

	it('renders with custom display name', () => {
		const props = { displayName: 'Display name text' };

		expect(LicenceSignIn).toRenderCorrectly(props);
	});

	it('renders if url is defined', () => {
		const props = { url: 'https://mytest.com' };

		expect(LicenceSignIn).toRenderCorrectly(props);
	});

});
