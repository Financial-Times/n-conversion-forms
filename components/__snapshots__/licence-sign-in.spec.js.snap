// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenceSignIn renders if url is defined 1`] = `
<div class="ncf__center">
  <div>
    <a href="https://mytest.com"
       class="ncf__button--secondary ncf__button--mid-width"
    >
      Sign in
    </a>
  </div>
</div>
`;

exports[`LicenceSignIn renders with custom display name 1`] = `
<div class="ncf__center">
  <p>
    <span>
      Have you already joined the
    </span>
    <span class="ncf__bold-licence-text">
      Display name text
    </span>
    <span>
      account?
    </span>
  </p>
</div>
`;

exports[`LicenceSignIn renders with default props 1`] = `
<div class="ncf__center">
</div>
`;
